File: backend/object-service/src/models/object.go
package models

type Object struct {
	ID          string `json:"id"`          // Unique identifier
	Name        string `json:"name"`        // Name of the object
	Type        string `json:"type"`        // Type of the object
	IsReserved  bool   `json:"isReserved"`  // Indicates if the object is reserved
	ReservedBy  string `json:"reservedBy"`  // User who reserved the object
    RoomID      string `json:"room_id"`     // ID of the room this object belongs to
}

File: frontend/src/types/object.ts
export interface Object {
  id: string;
  name: string;
  type: string;
  isReserved: boolean;
  reservedBy?: string;
  roomId: string;
}

export interface CreateObjectRequest {
  name: string;
  type: string;
  room_id: string;
}

export interface ReserveObjectRequest {
  userId: string;
  room_id: string;
}

export interface ObjectResponse {
  data: Object;
}

export interface ListObjectsResponse {
  data: Object[];
}